src/RA_Dlg_MemBookmark.cpp

```diff
@@ -821,51 +810,57 @@ void Dlg_MemBookmark::ImportFromFile(std::string sFilename)
 
 std::string Dlg_MemBookmark::ImportDialog()
 {
-    std::string str;
-
     if (g_pCurrentGameData->GetGameID() == 0)
     {
         MessageBox(nullptr, _T("ROM not loaded: please load a ROM first!"), _T("Error!"), MB_OK);
-        return str;
+        return "";
     }
-
-    CComPtr<IFileOpenDialog> pDlg;
-
-    HRESULT hr;
-    if (SUCCEEDED(hr = CoCreateInstance(CLSID_FileOpenDialog, nullptr, CLSCTX_ALL, IID_IFileOpenDialog, reinterpret_cast<void**>(&pDlg))))
+    
+    _CONSTANT_LOC BUF_SIZE{ 1024UL }; // Hopefully it's long enough, there's a fail-safe to prevent buffer overrun
+    
+    OPENFILENAME ofn{}; // doesn't seem to default construct by itself, a stack based equivalent to ZeroMemory
+    ofn.lStructSize  = static_cast<DWORD>(sizeof(OPENFILENAME));
+    ofn.hwndOwner    = m_hMemBookmarkDialog;
+    ofn.lpstrFilter  = c_rgFileTypes;
+    ofn.nFilterIndex = 1UL;
+    ofn.lpstrTitle   = _T("Import Bookmark File...");
+    ofn.nMaxFile     = BUF_SIZE - 1UL;
+    ofn.Flags        = OFN_ENABLESIZING | OFN_EXPLORER | OFN_PATHMUSTEXIST | OFN_FILEMUSTEXIST;
+    ofn.lpstrDefExt  = _T("txt");
+
+    ra::tstring sDefaultFilename;
+    sDefaultFilename.reserve(BUF_SIZE);
+    sDefaultFilename.append(ra::to_tstring(g_pCurrentGameData->GetGameID()));
+    sDefaultFilename.append(_T("-Bookmarks.txt"));
+    sDefaultFilename = sDefaultFilename.data();
```

Comment:
Building this string "normally" kept on causing heap corruption, so it's a bit non-standard but standard c++ complaint nonetheless.

Reply:
I think there was some buffer overrun, I'll put it back to ostringstream. The if block below should prevent a buffer overrun.

Reply:
I'm sorry dude it's still causing heap corruption when I want to use ostringstream, it's probably better the way I had it before.

Reply:
Please provide the `ostringstream` version you're having issues with. I can guess, but it would be best to know exactly.

Also, what is the purpose of line 851? `sDefaultFilename` should already be itself.

Reply:
Alright, For line 851 that's required or the length/size will be 0.
The only thing I can think of is the extra capacity reserved by the string, I'm gonna see if adding 16 to the buffer size does anything.

Reply:
Figured it out, had to make ofn.lpstrFile from OPENFILENAME saved as a static c buffer. Now there's no heap corruption. It works with ostringstream now. There's extra capacity allocated just incase.